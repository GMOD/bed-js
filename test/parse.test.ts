import { test, expect } from 'vitest'
import BED from '../src/index.ts'

test('BED3', () => {
  const p = new BED()
  const f1 = p.parseLine('contigA\t10875\t10884')
  const f2 = p.parseLine('co%2CtigA\t10875\t10884')
  const f3 = p.parseLine(['contigA', '10875', '10884'])
  expect(f1).toMatchSnapshot()
  expect(f2).toMatchSnapshot()
  expect(f3).toEqual(f1)
})

test('errors', () => {
  expect(() => new BED({ type: 'notexist' })).toThrow(/not found/)
})

test('BED6', () => {
  const p = new BED()
  const f1 = p.parseLine('contigA\t10875\t10884\ttest\t0.50\t-')
  const f2 = p.parseLine('co%2CtigA\t10875\t10884\ttest2\t0\t+')
  const f3 = p.parseLine('cotigA\t10875\t10884\ttest2\t.\t.')
  expect(f1).toMatchSnapshot()
  expect(f2).toMatchSnapshot()
  expect(f3).toMatchSnapshot()
  expect(f2.chrom).toEqual('co,tigA')
  expect(f2.chromStart).toEqual(10_875)
  expect(f2.chromEnd).toEqual(10_884)
  expect(f3.strand).toEqual(0)
})
test('BED12', () => {
  const p = new BED()
  const f1 = p.parseLine(
    'chr22\t1000\t5000\tcloneA\t960\t+\t1000\t5000\t0\t2\t567,488,\t0,3512',
  )
  const f2 = p.parseLine(
    'chr22\t2000\t6000\tcloneB\t900\t-\t2000\t6000\t0\t2\t433,399,\t0,3601',
  )
  expect(f1).toMatchSnapshot()
  expect(f2).toMatchSnapshot()
})

// this example comes from the LRA aligner preprint, it is BED format but the
// fields not really standard
test('BED weird', () => {
  const p = new BED()
  const line = `1	25608494	25737913	.	96.44058642	SRHB01000779.1	24831980	D`

  expect(p.parseLine(line)).toMatchSnapshot()
})

test('bigGenePred', () => {
  const p = new BED({ type: 'bigGenePred' })
  const f1 = p.parseLine(
    'chr1\t11868\t14409\tENST00000456328.2\t1000\t+\t11868\t11868\t255,128,0\t3\t359,109,1189,\t0,744,1352,\tDDX11L1\tnone\tnone\t-1,-1,-1,\tnone\tENST00000456328.2\tDDX11L1\tnone',
  )
  const f2 = p.parseLine(
    'chr1\t14403\t29570\tENST00000488147.1\t1000\t-\t14403\t14403\t255,128,0\t11\t98,34,152,159,198,136,137,147,99,154,37,\t0,601,1392,2203,2454,2829,3202,3511,3864,10334,15130,\tWASH7P\tnone\tnone\t-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,\tnone\tENST00000488147.1\tWASH7P\tnone',
  )
  expect(f1).toMatchSnapshot()
  expect(f2).toMatchSnapshot()
})

test('bigPsl', () => {
  const p = new BED({ type: 'bigPsl' })
  const f1 = p.parseLine(
    'chr1\t11873\t14361\tmAM992877\t1000\t+\t11873\t14361\t0\t3\t354,109,1141,\t0,739,1347,\t0\t1604\t+\t1604\t0,354,463,\t\t\t248956422\t1579\t25\t0\t0\t0',
  )
  const f2 = p.parseLine(
    'chr1\t11873\t14361\tmAM992877\t1000\t+\t12622\t13259\t0\t3\t354,109,1141,\t0,739,1347,\t0\t1604\t+\t1604\t0,354,463,\tcttgccgtcagccttttctttgacctcttctttctgttcatgtgtatctgctgtctcttagcccagacttcccgtgtcctttccaccgggcctttgggaggtcacagggtcttgatgctgtggtcttgatctgcaggtgtctgacttccagcaactgctggcctgtgccagggtggaagctgagcactggagtggagttttcctgtggagaggagccatgcctagagtgggatgggccattgttcatattctggcccctgttgtctgcatgtaacctaataccacgaccaggcatgggggaaagattggaggaaagttgagtgagaggatcaacttctctgacaacctaggccagtgtgtggtgatgccaggcatgcccttccccagcatcaggtctccagagctgcagaagacgacggccgacttggatcacactcttgtgagtgtccccagtgttgcagaggcagggccatcaggcaccaaagggattctgccagcatagtgctcctggaccagtgatacacccggcaccctgtcctggacaggctgttggcctggatctgagccctcgtggaggtcaaagccacctttggttctgccattactgctgtgtggaagttcactcctgccttttcctttcccgagagcctccaccaccccgagatcgcatttctcactgccttttgtctgcccagtttcaccagaagtaggcctcttcctgacaggcagctgcaccactgcctggcgctgcgcccttcctttgctctgcccgctggagacggtgtttgtcatgggcctggtctgcagggatcctgctacaaaggtgaaacccaggagagtgtggagtccagagtgttgccaggacccaggcacaggcattagtgcccgttggagaaaacaggggaatcccgaagaaatggtgggttttggccatccgtgagatcttcccagggcagctcccctctgtggaatccaatctgtcttccatcctgcgtggccgagggccaggcttctcactgggcctctgcaggaggctgccatttgtcctgcccaccgtcttagaagcgagacggagcagactcatctgctactgccctttctataataactaaagttagctgccctggactattcaccccctagtctcaatttaaaaagatccccatggccacagggcccctgcctgggggcttgtcacctcccccaccttcttcctgagtcacttctgcagccttgctccctaacctgccccacagccttgcctggatttctatctccctggcttggtgccagttcctccaagtcgatggcacctccctccctctcaaccacttgagcaaactccaagacatcttctaccccaacaccagcaattgtgccaagggccattaggctctcagcatgactatttttagagaccccgtgtctgtcactgaaaccttttttgtgggagactattcctcccatctgcaacagctgcccctgctgacggcccttctctcctccctctcatcccagagaaacaggtcagctgggagcttctgcccccactgcctagggaccaacaggggcaggaggcagtcactgaccccgagacgtttgca\t365..502\t248956422\t1579\t25\t0\t0\t1',
  )
  expect(f1).toMatchSnapshot()
  expect(f2).toMatchSnapshot()
})

test('bigMaf', () => {
  const p = new BED({ type: 'bigMaf' })
  const f1 = p.parseLine(
    'chr22_KI270731v1_random\t0\t2\ta score=60.000000;s hg38.chr22_KI270731v1_random        0 2 +   150754 AC;s canFam3.chr26                27974205 2 + 38964690 AT;',
  )
  expect(f1).toMatchSnapshot()
})

test('mafSummary', () => {
  const p = new BED({ type: 'mafSummary' })
  const f1 = p.parseLine(
    'chr22_KI270731v1_random\t0\t9270\tcanFam3\t0.568679\t\t',
  )
  expect(f1).toMatchSnapshot()
})

test('mafFrames', () => {
  const p = new BED({ type: 'mafFrames' })

  const f1 = p.parseLine(
    'chr22_KI270731v1_random\t11287\t12574\thg38\t0\t-\tENST00000619792.1\t-1\t-1\t1\t1',
  )
  expect(f1).toMatchSnapshot()
})

test('bigInteract', () => {
  const p = new BED({ type: 'bigInteract' })

  const f1 = p.parseLine(
    'chr3\t63702628\t63705638\t.\t584\t10\t.\t0\tchr17\t58878552\t58880897\t.\t.\tchr3\t63702628\t63705638\t.\t.',
  )
  expect(f1).toMatchSnapshot()
})

test('bigChain', () => {
  const p = new BED({ type: 'bigChain' })

  const f1 = p.parseLine(
    'chr22_KI270731v1_random\t133689\t133710\t530549\t1000\t-\t150754\tchr16\t98207768\t80144720\t80144725\t66',
  )
  expect(f1).toMatchSnapshot()
})

test('bigLink', () => {
  const p = new BED({ type: 'bigLink' })
  const f1 = p.parseLine('chr22_KI270731v1_random\t7917\t7918\t2912\t75185566')
  expect(f1).toMatchSnapshot()
})

test('bigNarrowPeak', () => {
  const p = new BED({ type: 'bigNarrowPeak' })

  expect(
    p.parseLine(
      'chr1\t566753\t566953\t.\t468\t.\t103.84\t5.54347\t4.80079\t154',
    ),
  ).toMatchSnapshot()
  expect(
    p.parseLine(
      'chr1\t566753\t566953\t.\t468\t.\t103.84\t5.54347\t4.80079\t154',
      { uniqueId: '1' },
    ),
  ).toMatchSnapshot()
})

test('real world', () => {
  const autoSql = `table hg18KGchr7
    "UCSC Genes for chr7 with color plus GeneSymbol and SwissProtID"
    (
    string  chrom;              "Reference sequence chromosome or scaffold"
    uint    chromStart; "Start position of feature on chromosome"
    uint    chromEnd;   "End position of feature on chromosome"
    string  name;               "Name of gene"
    uint    score;              "Score"
    char[1] strand;             "+ or - for strand"
    uint    thickStart; "Coding region start"
    uint    thickEnd;   "Coding region end"
    uint        reserved;       "Green on + strand, Red on - strand"
    string  geneSymbol; "Gene Symbol"
    string  spID;               "SWISS-PROT protein Accession number"
    )`
  const p = new BED({ autoSql })
  expect(p.autoSql).toMatchSnapshot()
})

test('gdc cancer', () => {
  const autoSql = `table gdcCancer
"somatic variants converted from MAF files obtained through the NCI GDC"
    (
    string chrom;      "Chromosome (or contig, scaffold, etc.)"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    string name;       "Name of item"
    uint   score;      "Score from 0-1000"
    char[1] strand;    "+ or -"
    uint thickStart;   "Start of where display should be thick (start codon)"
    uint thickEnd;     "End of where display should be thick (stop codon)"
    uint reserved;     "Used as itemRgb as of 2004-11-22"
    int blockCount;    "Number of blocks"
    int[blockCount] blockSizes; "Comma separated list of block sizes"
    int[blockCount] chromStarts; "Start positions relative to chromStart"
    string sampleCount;    "Number of samples with this variant"
    string freq;                    "Variant frequency"
    lstring Hugo_Symbol;            "Hugo symbol"
    lstring Entrez_Gene_Id;         "Entrez Gene Id"
    lstring Variant_Classification; "Class of variant"
    lstring Variant_Type;           "Type of variant"
    lstring Reference_Allele;       "Reference allele"
    lstring Tumor_Seq_Allele1;      "Tumor allele 1"
    lstring Tumor_Seq_Allele2;      "Tumor allele 2"
    lstring dbSNP_RS;               "dbSNP RS number"
    lstring dbSNP_Val_Status;       "dbSNP validation status"
    lstring days_to_death;          "Number of days till death"
    lstring cigarettes_per_day;     "Number of cigarettes per day"
    lstring weight;                 "Weight"
    lstring alcohol_history;        "Any alcohol consumption?"
    lstring alcohol_intensity;      "Frequency of alcohol consumption"
    lstring bmi;                    "Body mass index"
    lstring years_smoked;           "Number of years smoked"
    lstring height;                 "Height"
    lstring gender;                 "Gender"
    lstring project_id;             "TCGA Project id"
    lstring ethnicity;              "Ethnicity"
    lstring Tumor_Sample_Barcode;   "Tumor sample barcode"
    lstring Matched_Norm_Sample_Barcode;  "Matcheds normal sample barcode"
    lstring case_id;                "Case ID number"
)`
  const p = new BED({ autoSql })
  const returnValue = p.parseLine(
    'chr1\t1\t10\tT>G\t1\t.\t1815756\t1815757\t0,0,0\t1\t1\t0\t1\t0.0108695652174\tGNB1\t2782\tSplice_Region\tSNP\tT\tT\tG\tnovel\t\t--\t--\t--\t--\t--\t--\t--\t--\tfemale\tTCGA-ACC\tnot hispanic or latino\tTCGA-OR-A5KB-01A-11D-A30A-10\tTCGA-OR-A5KB-11A-11D-A30A-10\t09454ed6-64bc-4a35-af44-7c4344623d45',
  )
  expect(returnValue).toMatchSnapshot()
})
