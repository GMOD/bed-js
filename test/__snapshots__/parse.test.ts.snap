// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BED weird 1`] = `
{
  "chrom": "1",
  "chromEnd": 25737913,
  "chromStart": 25608494,
  "field5": "SRHB01000779.1",
  "field6": "24831980",
  "field7": "D",
  "name": ".",
  "score": 96.44058642,
  "strand": 0,
}
`;

exports[`BED3 1`] = `
{
  "chrom": "contigA",
  "chromEnd": 10884,
  "chromStart": 10875,
  "strand": 0,
}
`;

exports[`BED3 2`] = `
{
  "chrom": "co,tigA",
  "chromEnd": 10884,
  "chromStart": 10875,
  "strand": 0,
}
`;

exports[`BED6 1`] = `
{
  "chrom": "contigA",
  "chromEnd": 10884,
  "chromStart": 10875,
  "name": "test",
  "score": 0.5,
  "strand": -1,
}
`;

exports[`BED6 2`] = `
{
  "chrom": "co,tigA",
  "chromEnd": 10884,
  "chromStart": 10875,
  "name": "test2",
  "score": 0,
  "strand": 1,
}
`;

exports[`BED6 3`] = `
{
  "chrom": "cotigA",
  "chromEnd": 10884,
  "chromStart": 10875,
  "field4": ".",
  "field5": ".",
  "name": "test2",
  "strand": 0,
}
`;

exports[`BED12 1`] = `
{
  "blockCount": 2,
  "blockSizes": [
    567,
    488,
  ],
  "blockStarts": [
    0,
    3512,
  ],
  "chrom": "chr22",
  "chromEnd": 5000,
  "chromStart": 1000,
  "itemRgb": "0",
  "name": "cloneA",
  "score": 960,
  "strand": 1,
  "thickEnd": 5000,
  "thickStart": 1000,
}
`;

exports[`BED12 2`] = `
{
  "blockCount": 2,
  "blockSizes": [
    433,
    399,
  ],
  "blockStarts": [
    0,
    3601,
  ],
  "chrom": "chr22",
  "chromEnd": 6000,
  "chromStart": 2000,
  "itemRgb": "0",
  "name": "cloneB",
  "score": 900,
  "strand": -1,
  "thickEnd": 6000,
  "thickStart": 2000,
}
`;

exports[`bigChain 1`] = `
{
  "chainScore": 66,
  "chrom": "chr22_KI270731v1_random",
  "chromEnd": 133710,
  "chromStart": 133689,
  "name": "530549",
  "qEnd": 80144725,
  "qName": "chr16",
  "qSize": 98207768,
  "qStart": 80144720,
  "score": 1000,
  "strand": -1,
  "tSize": 150754,
}
`;

exports[`bigGenePred 1`] = `
{
  "blockCount": 3,
  "blockSizes": [
    359,
    109,
    1189,
  ],
  "cdsEndStat": "none",
  "cdsStartStat": "none",
  "chrom": "chr1",
  "chromEnd": 14409,
  "chromStart": 11868,
  "chromStarts": [
    0,
    744,
    1352,
  ],
  "exonFrames": [
    -1,
    -1,
    -1,
  ],
  "geneName": "ENST00000456328.2",
  "geneName2": "DDX11L1",
  "geneType": "none",
  "name": "ENST00000456328.2",
  "name2": "DDX11L1",
  "reserved": "255,128,0",
  "score": 1000,
  "strand": 1,
  "thickEnd": 11868,
  "thickStart": 11868,
  "type": "none",
}
`;

exports[`bigGenePred 2`] = `
{
  "blockCount": 11,
  "blockSizes": [
    98,
    34,
    152,
    159,
    198,
    136,
    137,
    147,
    99,
    154,
    37,
  ],
  "cdsEndStat": "none",
  "cdsStartStat": "none",
  "chrom": "chr1",
  "chromEnd": 29570,
  "chromStart": 14403,
  "chromStarts": [
    0,
    601,
    1392,
    2203,
    2454,
    2829,
    3202,
    3511,
    3864,
    10334,
    15130,
  ],
  "exonFrames": [
    -1,
    -1,
    -1,
    -1,
    -1,
    -1,
    -1,
    -1,
    -1,
    -1,
    -1,
  ],
  "geneName": "ENST00000488147.1",
  "geneName2": "WASH7P",
  "geneType": "none",
  "name": "ENST00000488147.1",
  "name2": "WASH7P",
  "reserved": "255,128,0",
  "score": 1000,
  "strand": -1,
  "thickEnd": 14403,
  "thickStart": 14403,
  "type": "none",
}
`;

exports[`bigInteract 1`] = `
{
  "chrom": "chr3",
  "chromEnd": 63705638,
  "chromStart": 63702628,
  "color": "0",
  "score": 584,
  "sourceChrom": "chr17",
  "sourceEnd": 58880897,
  "sourceStart": 58878552,
  "strand": 0,
  "targetChrom": "chr3",
  "targetEnd": 63705638,
  "targetStart": 63702628,
  "value": "10",
}
`;

exports[`bigLink 1`] = `
{
  "chrom": "chr22_KI270731v1_random",
  "chromEnd": 7918,
  "chromStart": 7917,
  "name": "2912",
  "qStart": 75185566,
  "strand": 0,
}
`;

exports[`bigMaf 1`] = `
{
  "chrom": "chr22_KI270731v1_random",
  "chromEnd": 2,
  "chromStart": 0,
  "mafBlock": "a score=60.000000;s hg38.chr22_KI270731v1_random        0 2 +   150754 AC;s canFam3.chr26                27974205 2 + 38964690 AT;",
  "strand": 0,
}
`;

exports[`bigNarrowPeak 1`] = `
{
  "chrom": "chr1",
  "chromEnd": 566953,
  "chromStart": 566753,
  "pValue": 5.54347,
  "peak": 154,
  "qValue": 4.80079,
  "score": 468,
  "signalValue": 103.84,
  "strand": 0,
}
`;

exports[`bigNarrowPeak 2`] = `
{
  "chrom": "chr1",
  "chromEnd": 566953,
  "chromStart": 566753,
  "pValue": 5.54347,
  "peak": 154,
  "qValue": 4.80079,
  "score": 468,
  "signalValue": 103.84,
  "strand": 0,
  "uniqueId": "1",
}
`;

exports[`bigPsl 1`] = `
{
  "blockCount": 3,
  "blockSizes": [
    354,
    109,
    1141,
  ],
  "chrom": "chr1",
  "chromEnd": 14361,
  "chromSize": 248956422,
  "chromStart": 11873,
  "chromStarts": [
    0,
    739,
    1347,
  ],
  "match": 1579,
  "misMatch": 25,
  "nCount": 0,
  "name": "mAM992877",
  "oCDS": "",
  "oChromEnd": 1604,
  "oChromSize": 1604,
  "oChromStart": 0,
  "oChromStarts": [
    0,
    354,
    463,
  ],
  "oSequence": "",
  "oStrand": "+",
  "repMatch": 0,
  "reserved": 0,
  "score": 1000,
  "seqType": 0,
  "strand": 1,
  "thickEnd": 14361,
  "thickStart": 11873,
}
`;

exports[`bigPsl 2`] = `
{
  "blockCount": 3,
  "blockSizes": [
    354,
    109,
    1141,
  ],
  "chrom": "chr1",
  "chromEnd": 14361,
  "chromSize": 248956422,
  "chromStart": 11873,
  "chromStarts": [
    0,
    739,
    1347,
  ],
  "match": 1579,
  "misMatch": 25,
  "nCount": 0,
  "name": "mAM992877",
  "oCDS": "365..502",
  "oChromEnd": 1604,
  "oChromSize": 1604,
  "oChromStart": 0,
  "oChromStarts": [
    0,
    354,
    463,
  ],
  "oSequence": "cttgccgtcagccttttctttgacctcttctttctgttcatgtgtatctgctgtctcttagcccagacttcccgtgtcctttccaccgggcctttgggaggtcacagggtcttgatgctgtggtcttgatctgcaggtgtctgacttccagcaactgctggcctgtgccagggtggaagctgagcactggagtggagttttcctgtggagaggagccatgcctagagtgggatgggccattgttcatattctggcccctgttgtctgcatgtaacctaataccacgaccaggcatgggggaaagattggaggaaagttgagtgagaggatcaacttctctgacaacctaggccagtgtgtggtgatgccaggcatgcccttccccagcatcaggtctccagagctgcagaagacgacggccgacttggatcacactcttgtgagtgtccccagtgttgcagaggcagggccatcaggcaccaaagggattctgccagcatagtgctcctggaccagtgatacacccggcaccctgtcctggacaggctgttggcctggatctgagccctcgtggaggtcaaagccacctttggttctgccattactgctgtgtggaagttcactcctgccttttcctttcccgagagcctccaccaccccgagatcgcatttctcactgccttttgtctgcccagtttcaccagaagtaggcctcttcctgacaggcagctgcaccactgcctggcgctgcgcccttcctttgctctgcccgctggagacggtgtttgtcatgggcctggtctgcagggatcctgctacaaaggtgaaacccaggagagtgtggagtccagagtgttgccaggacccaggcacaggcattagtgcccgttggagaaaacaggggaatcccgaagaaatggtgggttttggccatccgtgagatcttcccagggcagctcccctctgtggaatccaatctgtcttccatcctgcgtggccgagggccaggcttctcactgggcctctgcaggaggctgccatttgtcctgcccaccgtcttagaagcgagacggagcagactcatctgctactgccctttctataataactaaagttagctgccctggactattcaccccctagtctcaatttaaaaagatccccatggccacagggcccctgcctgggggcttgtcacctcccccaccttcttcctgagtcacttctgcagccttgctccctaacctgccccacagccttgcctggatttctatctccctggcttggtgccagttcctccaagtcgatggcacctccctccctctcaaccacttgagcaaactccaagacatcttctaccccaacaccagcaattgtgccaagggccattaggctctcagcatgactatttttagagaccccgtgtctgtcactgaaaccttttttgtgggagactattcctcccatctgcaacagctgcccctgctgacggcccttctctcctccctctcatcccagagaaacaggtcagctgggagcttctgcccccactgcctagggaccaacaggggcaggaggcagtcactgaccccgagacgtttgca",
  "oStrand": "+",
  "repMatch": 0,
  "reserved": 0,
  "score": 1000,
  "seqType": 1,
  "strand": 1,
  "thickEnd": 13259,
  "thickStart": 12622,
}
`;

exports[`gdc cancer 1`] = `
{
  "Entrez_Gene_Id": "2782",
  "Hugo_Symbol": "GNB1",
  "Matched_Norm_Sample_Barcode": "TCGA-OR-A5KB-11A-11D-A30A-10",
  "Reference_Allele": "T",
  "Tumor_Sample_Barcode": "TCGA-OR-A5KB-01A-11D-A30A-10",
  "Tumor_Seq_Allele1": "T",
  "Tumor_Seq_Allele2": "G",
  "Variant_Classification": "Splice_Region",
  "Variant_Type": "SNP",
  "alcohol_history": "--",
  "alcohol_intensity": "--",
  "blockCount": 1,
  "blockSizes": [
    1,
  ],
  "bmi": "--",
  "case_id": "09454ed6-64bc-4a35-af44-7c4344623d45",
  "chrom": "chr1",
  "chromEnd": 10,
  "chromStart": 1,
  "chromStarts": [
    0,
  ],
  "cigarettes_per_day": "--",
  "days_to_death": "--",
  "dbSNP_RS": "novel",
  "dbSNP_Val_Status": "",
  "ethnicity": "not hispanic or latino",
  "freq": "0.0108695652174",
  "gender": "female",
  "height": "--",
  "name": "T>G",
  "project_id": "TCGA-ACC",
  "reserved": "0,0,0",
  "sampleCount": "1",
  "score": 1,
  "strand": 0,
  "thickEnd": 1815757,
  "thickStart": 1815756,
  "weight": "--",
  "years_smoked": "--",
}
`;

exports[`mafFrames 1`] = `
{
  "chrom": "chr22_KI270731v1_random",
  "chromEnd": 12574,
  "chromStart": 11287,
  "frame": "0",
  "isExonEnd": "1",
  "isExonStart": "1",
  "name": "ENST00000619792.1",
  "nextFramePos": -1,
  "prevFramePos": -1,
  "src": "hg38",
  "strand": -1,
}
`;

exports[`mafSummary 1`] = `
{
  "chrom": "chr22_KI270731v1_random",
  "chromEnd": 9270,
  "chromStart": 0,
  "leftStatus": "",
  "rightStatus": "",
  "score": 0.568679,
  "src": "canFam3",
  "strand": 0,
}
`;

exports[`real world 1`] = `
{
  "comment": "UCSC Genes for chr7 with color plus GeneSymbol and SwissProtID",
  "fields": [
    {
      "arrayIsNumeric": undefined,
      "comment": "Reference sequence chromosome or scaffold",
      "isArray": undefined,
      "isNumeric": false,
      "name": "chrom",
      "type": "string",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Start position of feature on chromosome",
      "isArray": undefined,
      "isNumeric": true,
      "name": "chromStart",
      "type": "uint",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "End position of feature on chromosome",
      "isArray": undefined,
      "isNumeric": true,
      "name": "chromEnd",
      "type": "uint",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Name of gene",
      "isArray": undefined,
      "isNumeric": false,
      "name": "name",
      "type": "string",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Score",
      "isArray": undefined,
      "isNumeric": true,
      "name": "score",
      "type": "uint",
    },
    {
      "arrayIsNumeric": false,
      "comment": "+ or - for strand",
      "isArray": false,
      "isNumeric": false,
      "name": "strand",
      "size": 1,
      "type": "char",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Coding region start",
      "isArray": undefined,
      "isNumeric": true,
      "name": "thickStart",
      "type": "uint",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Coding region end",
      "isArray": undefined,
      "isNumeric": true,
      "name": "thickEnd",
      "type": "uint",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Green on + strand, Red on - strand",
      "isArray": undefined,
      "isNumeric": true,
      "name": "reserved",
      "type": "uint",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "Gene Symbol",
      "isArray": undefined,
      "isNumeric": false,
      "name": "geneSymbol",
      "type": "string",
    },
    {
      "arrayIsNumeric": undefined,
      "comment": "SWISS-PROT protein Accession number",
      "isArray": undefined,
      "isNumeric": false,
      "name": "spID",
      "type": "string",
    },
  ],
  "name": "hg18KGchr7",
  "type": "table",
}
`;
